modules/core/src/smithy4s/internals/DocumentEncoderSchemaVisitor.scala

```diff
@@ -101,7 +101,7 @@ class DocumentEncoderSchemaVisitor(
         .getOrElse(TimestampFormat.EPOCH_SECONDS) match {
         case DATE_TIME     => ts => DString(ts.format(DATE_TIME))
         case HTTP_DATE     => ts => DString(ts.format(HTTP_DATE))
-        case EPOCH_SECONDS => ts => DNumber(BigDecimal(ts.epochSecond))
+        case EPOCH_SECONDS => ts => DNumber(BigDecimal(s"${ts.epochSecond}.${ts.nano}"))
```

Comment:
**thought:** I am not really a fan of constructing bigDecimal from string in here but I couldn't think of anything better.

Reply:
BigDecimal(ts.epochSecond + 10e-9 * ts.nano) ?

Reply:
I should've made myself clearer. I thought about this approach, but I didn't want to use because of potential precision lose issue. 

Consider following:
```scala
val now = Instant.now()
val epochSecond: Long = now.getEpochSecond
val nano: Int = 999_999_999 // almost 1 second

val result: Double = epochSecond + 10e-9 * nano
println(s"Result with potential precision loss: ${BigDecimal(result)}")

val accurateBigDecimal =
  BigDecimal(epochSecond) + (BigDecimal(nano) * BigDecimal(10).pow(-9))

println(s"Accurate result : ${accurateBigDecimal}")

```
Which prints:
```
Result with potential precision loss: 1725269367
Accurate result : 1725269357.999999999
```